test/integration/kubernetes_deploy_test.rb

```diff
@@ -806,6 +806,46 @@ def test_invalid_context
     end
   end
 
+  def test_multiple_configuration_files
+    old_config = ENV['KUBECONFIG']
+    begin
+      config_file = File.join(__dir__, '../fixtures/kube-config/unknown_config.yml')
+      ENV['KUBECONFIG'] = config_file
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
+        'Result: FAILURE',
+        'Configuration invalid',
+        "Kube config not found at #{config_file}"
+      ], in_order: true)
+
+      ENV['KUBECONFIG'] = " : "
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
+        'Result: FAILURE',
+        'Configuration invalid',
+        "Kube config file name(s) not set in $KUBECONFIG"
+      ], in_order: true)
+
+      ENV['KUBECONFIG'] = nil
```

Comment:
FYI this case is also covered by `test_error_message_when_kubeconfig_not_set` in `test/unit/kubernetes-deploy/deploy_task_test.rb`. Technically most of your new test could be split into separate unit tests in that file, since only the successful case needs to be able to hit the api if I'm not mistaken. I'm fine either way though.

Reply:
This new test function is focused on ENV KUBECONFIG. It does not verify other mis config cases which were covered in test_error_message_when_kubeconfig_not_set. Besides, it tested multi-config in integrated test case, so I would prefer to leave it in kubernetes_deploy_test.